src/plan.rs

```diff
@@ -191,6 +193,52 @@ impl Recipe {
             Recipe::BluesteinsAlgorithm { len, .. } => *len,
         }
     }
+
+    pub fn cost(&self, repeats: usize) -> f32 {
```

Comment:
One thing that I'd like to change at some point in the future, but we don't need to address for this PR, is that the current API for estimating costs doesn't allow for any kind of memoization or caching of costs. Every time you want to estimate the cost of a recipe, you have to walk the entire tree of that recipe. If there was a way we could make cost() just return a cached value, we could check the cost of many, many more recipes without worrying how much time it takes.

We don't need to fix it for this PR, but something to think about.